File: src/main/java/net/thedragonteam/thedragonlib/client/ResourceHelperDL.java
package net.thedragonteam.thedragonlib.client;

import net.minecraft.client.Minecraft;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import net.thedragonteam.thedragonlib.TheDragonLib;

import java.util.HashMap;
import java.util.Map;

public class ResourceHelperDL {

    public static final String RESOURCE_PREFIX = TheDragonLib.MODID + ":";
    private static ResourceLocation vanillaParticles;
    private static Map<String, ResourceLocation> cachedResources = new HashMap<>();

    @SideOnly(Side.CLIENT)
    public static void bindTexture(ResourceLocation texture) {
        Minecraft.getMinecraft().renderEngine.bindTexture(texture);
    }

    public static ResourceLocation getResource(String rs) {
        if (!cachedResources.containsKey(rs)) cachedResources.put(rs, new ResourceLocation(RESOURCE_PREFIX + rs));
        return cachedResources.get(rs);
    }

    public static ResourceLocation getResourceRAW(String rs) {
        if (!cachedResources.containsKey(rs)) cachedResources.put(rs, new ResourceLocation(rs));
        return cachedResources.get(rs);
    }

    public static void bindTexture(String rs) {
        bindTexture(getResource(rs));
    }
}

File: src/main/java/net/thedragonteam/thedragonlib/client/ResourceHelperDL.kt
package net.thedragonteam.thedragonlib.client

import net.minecraft.client.Minecraft
import net.minecraft.util.ResourceLocation
import net.thedragonteam.thedragonlib.TheDragonLib
import java.util.*

object ResourceHelperDL {

    val RESOURCE_PREFIX = TheDragonLib.MODID + ":"
    private val cachedResources = HashMap<String, ResourceLocation>()

    fun bindTexture(texture: ResourceLocation) {
        Minecraft.getMinecraft().renderEngine.bindTexture(texture)
    }

    fun getResource(rs: String): ResourceLocation {
        if (!cachedResources.containsKey(rs)) cachedResources.put(rs, ResourceLocation(RESOURCE_PREFIX + rs))
        return cachedResources[rs] as ResourceLocation
    }

    fun getResourceRAW(rs: String): ResourceLocation {
        if (!cachedResources.containsKey(rs)) cachedResources.put(rs, ResourceLocation(rs))
        return cachedResources[rs] as ResourceLocation
    }

    fun bindTexture(rs: String) {
        bindTexture(getResource(rs))
    }
}